### PR TITLE
Add HotDox

### PR DESCRIPTION
**Pull request by layne001365**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by layne001365** [comment]
Add HotDox to repo.

File: keyboards/hotdox/config.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LCTL) | MOD_BIT(KC_RCTL)) || \
+    keyboard_report->mods == (MOD_BIT(KC_LSFT) | MOD_BIT(KC_RSFT)) \
```
Comment (vomindoraan) [inline keyboards/hotdox/config.h]: Please make the following changes to this piece of code:
* Replace `keyboard_report->mods` with `get_mods()`.
* Replace `(MOD_BIT(KC_LCTL) | MOD_BIT(KC_RCTL))` with `MOD_MASK_CTRL`.
* Replace `(MOD_BIT(KC_LCTL) | MOD_BIT(KC_RCTL))` with `MOD_MASK_SHIFT`.

Comment (drashna) [reply keyboards/hotdox/config.h]: This. 

Otherwise, some of the recent changes cause the `keyboard_report` structure to not be completely accurate.   To ensure that this works, you'd need to make these changes.

File: keyboards/hotdox/config.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#ifndef ERGODOX_HOTDOX_CONFIG_H
```
Comment (drashna) [inline keyboards/hotdox/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch? 

```suggestion
#pragma once
```

File: keyboards/hotdox/config.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+#define ERGODOX_HOTDOX_CONFIG_H
```
Comment (drashna) [inline keyboards/hotdox/config.h]:
```suggestion
```

File: keyboards/hotdox/config.h [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+//#define NO_ACTION_MACRO
+//#define NO_ACTION_FUNCTION
+#define DEBUG_MATRIX_SCAN_RATE
+
+#endif
```
Comment (drashna) [inline keyboards/hotdox/config.h]:
```suggestion
```

File: keyboards/hotdox/config.h [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+/* disable print */
+// #define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+//#define NO_ACTION_MACRO
+//#define NO_ACTION_FUNCTION
+#define DEBUG_MATRIX_SCAN_RATE
```
Comment (drashna) [inline keyboards/hotdox/config.h]: Is this needed?

File: keyboards/hotdox/config.h [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+#define LED_BRIGHTNESS_LO       15
+#define LED_BRIGHTNESS_HI       255
+
+/* fix space cadet rollover issue */
+#define DISABLE_SPACE_CADET_ROLLOVER
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE    5
+
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/hotdox/config.h]: This setting should be removed, as it's no longer used.

Specifically, it's the new default behavior for QMK Firmware.  And you can disable it by defining `STRICT_LAYER_RELEASE` instead. 

```suggestion
```

File: keyboards/hotdox/i2cmaster.h [hunk 7]
```diff
@@ -0,0 +1,178 @@
+#ifndef _I2CMASTER_H
```
Comment (drashna) [inline keyboards/hotdox/i2cmaster.h]: Is this actually needed?

File: keyboards/hotdox/info.json [hunk 8]
```diff
@@ -0,0 +1,107 @@
+{
+    "keyboard_name": "Ergodone",
```
Comment (drashna) [inline keyboards/hotdox/info.json]: Not sure this is right. :)

File: keyboards/hotdox/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,222 @@
+#include QMK_KEYBOARD_H
+#include "debug.h"
```
Comment (drashna) [inline keyboards/hotdox/keymaps/default/keymap.c]: This line is not actually needed, and should be removed. 

```suggestion
```

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/hotdox/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+#include "action_layer.h"
```
Comment (drashna) [inline keyboards/hotdox/keymaps/default/keymap.c]: This line is not actually needed, and should be removed. 

```suggestion
```

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/hotdox/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+#include "version.h"
+
+#define BASE 0 // default layer
+#define SYMB 1 // symbols
+#define MDIA 2 // media keys
+
+enum custom_keycodes {
+  PLACEHOLDER = SAFE_RANGE, // can always be here
+  EPRM,
```
Comment (drashna) [inline keyboards/hotdox/keymaps/default/keymap.c]: This can be removed, as `EEP_RST` handles this.

File: keyboards/hotdox/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+                 KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MPLY,
+       KC_TRNS,  KC_TRNS, KC_TRNS, KC_MPRV, KC_MNXT, KC_TRNS, KC_TRNS,
+                          KC_VOLU, KC_VOLD, KC_MUTE, KC_TRNS, KC_TRNS,
+       KC_TRNS, KC_TRNS,
+       KC_TRNS,
+       KC_TRNS, KC_TRNS, KC_WBAK
+),
+};
+
+const uint16_t PROGMEM fn_actions[] = {
```
Comment (drashna) [inline keyboards/hotdox/keymaps/default/keymap.c]: Could you remove the `fn_actions` block?  All of it's features have been added as keycodes, and been deprecated in favor of `process_record_user`.

File: keyboards/hotdox/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+    [1] = ACTION_LAYER_TAP_TOGGLE(SYMB)                // FN1 - Momentary Layer 1 (Symbols)
+};
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt)
```
Comment (drashna) [inline keyboards/hotdox/keymaps/default/keymap.c]: Could you remove the `action_get_macro` function?  It's been deprecated in favor of `process_record_user`.

File: keyboards/hotdox/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,222 @@
...
+        }
+        break;
+      }
+    return MACRO_NONE;
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    // dynamically generate these.
+    case EPRM:
```
Comment (drashna) [inline keyboards/hotdox/keymaps/default/keymap.c]: Could you remove this macro? 

This exists in the core as `EEP_RST`

File: keyboards/hotdox/left.c [hunk 15]
```diff
@@ -0,0 +1,140 @@
+#include <stdbool.h>
```
Comment (drashna) [inline keyboards/hotdox/left.c]: Why is this file needed?

File: keyboards/hotdox/rules.mk [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+#
+# For advanced users only:
+# make teensy = Download the hex file to the device, using teensy_loader_cli.
+#               (must have teensy_loader_cli installed).
+#
+#----------------------------------------------------------------------------
+
+# # project specific files
+SRC =  matrix.c     \
+	   twimaster.c  \
```
Comment (drashna) [inline keyboards/hotdox/rules.mk]: Please remove this and replace it with i2c_master.c, which is the global implementation of this. 

```suggestion
	   i2c_master.c  \
```

File: keyboards/hotdox/twimaster.c [hunk 17]
```diff
@@ -0,0 +1,210 @@
+/*************************************************************************
```
Comment (drashna) [inline keyboards/hotdox/twimaster.c]: Please remove this file.

**Comment by drashna** [comment]
Travis CI is failing because `ergodox_board_led_on` is missing, and a lot of boards expect it.

Also, you may want to check the Ergodox EZ, as it looks like this code is based on it, but is out of date.
